File: DesignPatternsExercises/java/MediatorDesignPattern/Solution/Buyer.java
public class Buyer {
	private Mediator mediator;
	private String currency;

	public Buyer(Mediator mediator, String currency) {
		this.mediator = mediator;
		this.currency = currency;
		// mediator.__________(1)__________;
		mediator.addBuyer(this);
	}

	// public void __________(2)__________(float bid) {
	public void attemptToPurchase(float bid) {
		System.out.println("Buyer is attempting a bid of " + currency + " " + bid);
		mediator.placeBid(currency, bid);
	}
}

File: DesignPatternsExercises/ruby/mediator_design_pattern/solution/buyer.rb
class Buyer
  def initialize(mediator, currency)
    @mediator = mediator
    @currency = currency
    # mediator.__________(1)__________
    mediator.add_buyer(self)
  end

  # def __________(2)__________(bid)
  def attempt_to_purchase(bid)
    puts "Buyer is attempting a bid of #{@currency} #{bid}"
    @mediator.place_bid(@currency, bid)
  end
end
